### ISSUE TITLE
Support different index types within MultiIndex

### ISSUE THREAD

**Issue by v0dro** [v0dro/daru#150]
Currently MultiIndex is quite limited in functionality. It does provide hierarchical indexing, but this is quite limited to having simple ruby objects for representing each level. It would be great to say, have a `DateTimeIndex` on the second level of an hierarchy for selecting elements by date on the second level and by normal indexing on the first level (the first level will be a `Daru::Index`).

In short it should be able to support different Indexing classes on each level.
